realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -542,17 +553,24 @@ public void onChange(RealmResults <Permission> loadedPermissions) {
                 });
             }
         }
+
+        private void notifyCallbackWithSuccess(RealmResults<Permission> permissions) {
+            callback.onSuccess(permissions);
+            activeTasks.remove(this);
```

Comment:
Yes, it originally lived there, but with multiple different callbacks, it became impossible to keep there.